src/main/java/org/threadly/concurrent/future/SettableListenableFuture.java

```diff
@@ -187,6 +241,8 @@ public T get() throws InterruptedException, ExecutionException {
       
       if (failure != null) {
         throw new ExecutionException(failure);
```

Comment:
yeah I guess if thats how the java.util.concurrent.Future does it then ok, but I personally think it would work much better, your essentially telling them the Execution failed for the reason of getting cancled, which makes perfect since to me and takes care of the unchecked Runnable issue.